Wrappers/Python/ccpi/optimisation/algorithms/Algorithm.py

```diff
@@ -183,26 +183,44 @@ def run(self, iterations=None, verbose=True, callback=None, very_verbose=False):
         i = 0
         if verbose:
             print (self.verbose_header(very_verbose))
-        if self.iteration == 0:
-            if verbose:
-                print(self.verbose_output(very_verbose))
+        
         for _ in self:
+            # __next__ is called
+
+            # the following code is just for displaying purposes of the status of the minimisation
+
+            # self.iteration is incremented in __next__, so now we have 
+            # self.iteration is one iteration larger than what we want to display
+            self.iteration -= 1
```

Comment:
Τhis is a bit strange that you need to reduce it by 1 and the increase it by one after you get the objective.

Reply:
`self.iteration` gets incremented in `__next__` which is not explicitly called but it's the first call that happens in the [`for`](https://github.com/vais-ral/CCPi-Framework/blob/f4b28f673bcd34280f25887b98f0265bb429085f/Wrappers/Python/ccpi/optimisation/algorithms/Algorithm.py#L188) loop. 

This means that the `print` statements will access the iteration number which 1 larger than what the print would refer to.
 
Instead of changing all the prints to reduce the iteration number I just reduced it at the beginning of the "print" section and increased it to the proper value at the end.

Reply:
Ok, I cannot find any other way right now. If we're all happy, let's merge it because it's important.